### DISCUSSION TITLE
Negative R2 for some models for regression models

### DISCUSSION THREAD

**Discussion by psrana** [pycaret/pycaret#1711] [Q&A]
**Bug**
Negative R2 for some models for regression models while working with boston dataset

![bug](https://user-images.githubusercontent.com/7460892/137583630-c77f2dd2-951f-41a5-8530-efe5ddf919e0.png)

**Code**
```python
from pycaret.datasets import get_data
from pycaret.regression import *

bostonDataSet= get_data("boston")
s = setup(data = bostonDataSet, target='medv', silent=True)
cm = compare_models()
```

**Expected behavior**
R2 (R-Square) value must be positive

**Comment by moezali1** [comment]
@psrana Not a bug in PyCaret. This is possible when the model is a poor fit.

https://stats.stackexchange.com/questions/183265/what-does-negative-r-squared-mean

**Reply by Andreas70180** [reply]
I got the same problem. But i only get negative r-values. Going from -0,5 to over -10. This shouldn´t be the case, right?
